core/src/test/kotlin/org/stellar/anchor/sep24/Sep24ServiceTest.kt

```diff
@@ -274,6 +328,8 @@ internal class Sep24ServiceTest {
     assertEquals(TEST_ASSET_ISSUER_ACCOUNT_ID, slotTxn.captured.requestAssetIssuer)
     assertEquals(TEST_ACCOUNT, slotTxn.captured.toAccount)
     assertEquals(TEST_CLIENT_DOMAIN, slotTxn.captured.clientDomain)
+    assertEquals(depositQuote.id, slotTxn.captured.quoteId)
+    assertEquals(depositQuote.sellAsset, slotTxn.captured.sourceAsset)
```

Comment:
Can we add more unit tests for the actual quote validation? I think you'll find it easier to test if the `validatedAndPopulateQuote` is in its own class. I'll leave that decision up to you, though.